packages/workflow/src/WorkflowDataProxy.ts

```diff
@@ -159,6 +160,84 @@ export class WorkflowDataProxy {
 		);
 	}
 
+	private buildAgentToolInfo(node: INode) {
+		const nodeType = this.workflow.nodeTypes.getByNameAndVersion(node.type, node.typeVersion);
+		const type = nodeType.description.displayName;
+		const params = NodeHelpers.getNodeParameters(
+			nodeType.description.properties,
+			node.parameters,
+			true,
+			false,
+			node,
+			nodeType.description,
+		);
+		const resourceKey = params?.resource;
+		const operationKey = params?.operation;
+
+		const resource =
+			nodeType.description.properties
+				.find((x) => x.name === 'resource')
+				?.options?.find((x) => 'value' in x && x.value === resourceKey)?.name ?? null;
+
+		const operation =
+			nodeType.description.properties
+				.find(
+					(x) =>
+						x.name === 'operation' &&
+						x.displayOptions?.show?.resource?.some((y) => y === resourceKey),
+				)
+				?.options?.find((y) => 'value' in y && y.value === operationKey)?.name ?? null;
+
+		const hasCredentials = !isObjectEmpty(node.credentials ?? {});
+
+		const hasValidCalendar = nodeType.description.name.includes('googleCalendar')
+			? isResourceLocatorValue(node.parameters.calendar) && node.parameters.calendar.value !== ''
+			: undefined;
+
+		const aiDefinedFields = Object.entries(node.parameters)
+			.map(([key, value]) => [key, isResourceLocatorValue(value) ? value.value : value] as const)
+			.filter(([_, value]) => value?.toString().toLowerCase().includes('$fromai'))
+			.map(([key]) => nodeType.description.properties.find((x) => x.name === key)?.displayName);
+
+		return {
+			name: node.name,
+			type,
+			resource,
+			operation,
+			hasCredentials,
+			hasValidCalendar,
+			aiDefinedFields,
+		};
+	}
+
+	private agentInfo() {
+		const node = this.workflow.getNode(this.activeNodeName);
+		if (!node || node.type !== AGENT_LANGCHAIN_NODE_TYPE) return undefined;
+		const connectedTools = this.workflow
+			.getParentNodes(this.activeNodeName, NodeConnectionTypes.AiTool)
+			.map((x) => this.workflow.getNode(x))
+			.filter((x) => x) as INode[];
+		const memoryConnectedToAgent =
+			this.workflow.getParentNodes(this.activeNodeName, NodeConnectionTypes.AiMemory).length > 0;
+		const allTools = this.workflow.queryNodes((x) => {
+			return x.description.name.toLowerCase().includes('tool');
+		});
+
+		const unconnectedTools = allTools
+			.filter(
+				(x) => this.workflow.getChildNodes(x.name, NodeConnectionTypes.AiTool, 1).length === 0,
+			)
+			.filter((x) => !connectedTools.includes(x));
+
+		return {
+			memoryConnectedToAgent,
+			tools: [
+				...connectedTools.map((x) => ({ connected: true, ...this.buildAgentToolInfo(x) })),
```

Comment:
@CharlieKolb We should avoid using `x` as a name for parameters or variables. It would be clearer to use the actual name for what the property is